pkg/modules/dnsnames.go

```diff
@@ -0,0 +1,80 @@
+package modules
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/go-ldap/ldap/v3"
+	"github.com/ropnop/go-windapsearch/pkg/ldapsession"
+	"github.com/spf13/pflag"
+)
+
+type DnsNamesModule struct{}
+
+func init() {
+	AllModules = append(AllModules, new(DnsNamesModule))
+}
+
+func (d DnsNamesModule) Name() string {
+	return "dns-names"
+}
+
+func (d DnsNamesModule) Description() string {
+	return "List all DNS Names"
+}
+
+func (d DnsNamesModule) FlagSet() *pflag.FlagSet {
+	flags := pflag.NewFlagSet(d.Name(), pflag.ExitOnError)
+	return flags
+}
+
+func (d DnsNamesModule) DefaultAttrs() []string {
+	return []string{"dn"}
```

Comment:
Do you happen to know if requesting an attribute we don't have permission for triggers anything? I don't see a problem with adding some additional attrs like "name" here since if you don't have permission they just won't return, but I wouldn't want to trigger alerts if that's the case

Reply:
I don't know, I left name out because it is often empty and might be confusing to some users. We can show the name and document that it is empty, but I still see many people overlooking this and relying on name then. We may consider parsing the dn to a regular FQDN and returning this also, because that's actually why the people will be here

Reply:
If it's empty it just won't show in the JSON results, which I think is okay. DN will always be there. Dirkjan requests `name`, `dnsRecord`, and `dnsTombstoned` by default, which makes sense to me. I'll just have to write a JSON marshaler for the dnsRecord binary format, which sounds like it would be really useful anyway :)